### ISSUE TITLE
Two distributor metrics are reported incorrectly

### ISSUE THREAD

**Issue by replay** [grafana/mimir#2784]
#### Describe the bug

The following two metrics:

```
distributor_samples_in_total
distributor_exemplars_in_total
```

Are supposed to also include samples/exemplars that got dropped by the HA-deduplication or relabeling rules, but currently they don't because they get updated only after these two steps have been performed.

#### Expected behavior

These metrics should also account for samples/examplars that get dropped.

#### Additional Context

This is a result of the refactoring which moved multiple parts of `Distributor.PushWithCleanup()` into separate middlewares.

**Comment by pracucci** [comment]
Looks like a duplicate of #2783

**Comment by replay** [comment]
>Looks like a duplicate of https://github.com/grafana/mimir/issues/2783

It is, closing
